pymc/distributions/distribution.py

```diff
@@ -687,6 +747,7 @@ def custom_dist_logcdf(op, value, size, *params, **kwargs):
 
         @_moment.register(rv_type)
         def custom_dist_get_moment(op, rv, size, *params):
+            params = [i for i in params if not isinstance(i, RandomGeneratorSharedVariable)]
```

Comment:
Why is this needed?

Reply:
`params` consists `RandomGeneratorSharedVariable`s, which are not match with `dist` function signature. So I filter out it here, but may be there is a better way to do it

Reply:
Ah, I think belowe we do something like `params[:len(dist_params)]`? Does that work? Would be nice to add a comment to say we are excluding the shared RNGs

Reply:
I think `params[:len(dist_params)]` should work here, but do we have access to `dist_params` from this function? Looks like no.

Reply:
In that case let's keep like you did, but perhaps use a helper function with a readable name? Could that helper be used below?

Reply:
I added helper function to filter out shared RNGs